source/screens/caseScreens/CaseSummary.tsx

```diff
@@ -354,6 +372,42 @@ const CaseSummary = (props) => {
     void updateCaseAfterSignature();
   }, [updateCaseSignature, authContext.status, caseId, getCase]);
 
+  const getPasswordAccumulator = useCallback(
+    async (
+      oldValue,
+      caseItem,
+      caseUser: UserInterface
+    ): Promise<{ password: string }> => {
+      const { currentFormId } = caseItem;
+      const form = caseItem.forms[currentFormId];
+      const hasSymmetricKey = !!form.encryption.symmetricKeyName;
+
+      const encryptionPin = hasSymmetricKey
+        ? await getPasswordForForm(form, caseUser)
+        : undefined;
+
+      return { ...oldValue, [currentFormId]: encryptionPin };
+    },
+    []
+  );
+
+  useEffect(() => {
+    const tryGetPassword = async () => {
+      if (Object.keys(cases).length > 0) {
+        const getPasswordsPromise = Object.values(cases).reduce(
+          (oldValue, newValue) =>
+            getPasswordAccumulator(oldValue, newValue, authContext.user),
+          Promise.resolve({})
+        );
+
+        const formPasswords = await getPasswordsPromise;
+        setPasswords(formPasswords);
+      }
+    };
+
+    void tryGetPassword();
+  }, [cases, authContext.user, getPasswordAccumulator]);
+
```

Comment:
This code is now in both `caseOverview` and `caseSummary`. I suggest that you make a custom hook for this and re-use on both screens.


**useGetFormPasswords.ts**
```js
import { useEffect, useState, useCallback } from "react";

import { getPasswordForForm } from "../../services/encryption/CaseEncryptionHelper";

import type { Case } from "../../types/Case";
import type { User } from "../../types/UserTypes";

type Password = Record<string, string | undefined>;

function useGetFormPasswords(
  cases: Record<string, Case>,
  user: User | null
): Password {
  const [passwords, setPasswords] = useState<Password>({});

  const getPasswordAccumulator = useCallback(
    async (
      oldValue,
      caseItem,
      caseUser: User
    ): Promise<{ password: string }> => {
      const { currentFormId } = caseItem;
      const form = caseItem.forms[currentFormId];
      const hasSymmetricKey = !!form.encryption.symmetricKeyName;

      const encryptionPin = hasSymmetricKey
        ? await getPasswordForForm(form, caseUser)
        : undefined;

      return { ...oldValue, [currentFormId]: encryptionPin };
    },
    []
  );

  useEffect(() => {
    const tryGetPassword = async () => {
      if (Object.keys(cases).length > 0 && user !== null) {
        const getPasswordsPromise = Object.values(cases).reduce(
          (oldValue, newValue) =>
            getPasswordAccumulator(oldValue, newValue, user),
          Promise.resolve({})
        );

        const formPasswords = await getPasswordsPromise;
        setPasswords(formPasswords as Password);
      }
    };

    void tryGetPassword();
  }, [cases, user, getPasswordAccumulator]);

  return passwords;
}

export default useGetFormPasswords;

```

**CaseSummary/CaseOverview**

```js
...
import useGetFormPasswords from "./useGetFormPasswords";
...

function CaseOverview/Summary() {
  ...
 const passwords = useGetFormPasswords(cases, user)

}
```

Reply:
Fixed, ty!